server/src/linker.js

```diff
@@ -42,3 +42,13 @@ exports.imageLink = function (urlBase, resource) {
   }
   return urlBase + "/images" + resource;
 };
+
+exports.contentLink = function (urlBase, resource) {
+  if (! resource.startsWith("/")) {
+    resource = "/" + resource;
+  }
+  if (resource.startsWith("/content")) {
+    throw new Error("imageLink URL should not start with /content: " + resource);
+  }
+  return urlBase + "/content" + resource;
+};
```

Comment:
Things like viewUrl are already on the shot, and the shot already knows both the base and its name.